File: frontend/src/types/inventory.ts
export type ProductUnit = 'UN' | 'CX' | 'KG' | 'G' | 'L' | 'ML' | 'M' | 'CM'

export type Category = {
  description: string | null
  id: number
  is_active: boolean
  name: string
}

export type Product = {
  category_id: number
  category_name: string
  id: number
  is_active: boolean
  is_below_minimum: boolean
  minimum_stock: string
  name: string
  quantity: string
  sku: string
  unit: ProductUnit
}

export type CategoryPayload = {
  description?: string | null
  is_active?: boolean
  name: string
}

export type ProductPayload = {
  category_id: number
  is_active?: boolean
  minimum_stock: string
  name: string
  sku: string
  unit: ProductUnit
}

File: backend/app/schemas/inventory.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.inventory import ProductUnit


class CategoryCreate(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    description: str | None = Field(default=None, max_length=500)
    is_active: bool = True


class CategoryUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    description: str | None = Field(default=None, max_length=500)
    is_active: bool | None = None


class CategoryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: str | None
    is_active: bool


class ProductCreate(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    sku: str = Field(min_length=1, max_length=64)
    category_id: int
    unit: ProductUnit
    minimum_stock: Decimal = Field(ge=0, max_digits=12, decimal_places=3)
    is_active: bool = True


class ProductUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    sku: str | None = Field(default=None, min_length=1, max_length=64)
    category_id: int | None = None
    unit: ProductUnit | None = None
    minimum_stock: Decimal | None = Field(
        default=None,
        ge=0,
        max_digits=12,
        decimal_places=3,
    )
    is_active: bool | None = None


class ProductResponse(BaseModel):
    id: int
    name: str
    sku: str
    category_id: int
    category_name: str
    unit: ProductUnit
    minimum_stock: Decimal
    quantity: Decimal
    is_active: bool
    is_below_minimum: bool
